framework/src/modules/chain/dpos/dpos.js

```diff
@@ -1,11 +1,27 @@
-const Delegates = require('./delegates');
+const { Delegates, EVENT_ROUND_FINISHED } = require('./delegates');
 
 module.exports = class Dpos {
-	constructor({ storage, logger }) {
+	constructor({ storage, logger, slots }) {
+		this.finalizedBlockRound = 0;
 		this.delegate = new Delegates({ storage, logger });
+		this.slots = slots;
+		this.delegate.on(EVENT_ROUND_FINISHED, () => {
+			this.onRoundFinish();
+		});
 	}
 
 	async getRoundDelegates(round) {
 		return this.delegates.getRoundDelegates(round);
 	}
+
+	async onBlockFinalized({ height }) {
```

Comment:
I think the naming of this function should be more descriptive.
rather than calling it `onBlockFinalized` maybe `updateFinalizedRound`?

Reply:
It is going to be used as an event handler.